code/src/test/kotlin/com/expediagroup/sdk/core/http/RequestTest.kt

```diff
@@ -0,0 +1,286 @@
+
+        assertEquals("application/json", originalRequest.headers.get("Content-Type"))
+        assertNull(originalRequest.headers.get("Accept"))
+
+        assertEquals("application/json", newRequest.headers.get("Content-Type"))
+        assertEquals("text/plain", newRequest.headers.get("Accept"))
+    }
+
+    @Test
+    fun `should build a request with valid method and url string`() {
+        // Given
+        val urlString = "https://example.com"
+
+        // When
+        val request = Request
+            .builder()
+            .method(Method.GET)
+            .url(urlString)
+            .build()
+
+        // Expect
+        assertEquals(URL(urlString), request.url)
+    }
+
+    @Test
+    fun `should add single header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue = "application/json"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .addHeader(headerName, headerValue)
+            .build()
+
+        // Expect
+        assertEquals(headerValue, request.headers.get(headerName))
+    }
+
+    @Test
+    fun `should add multiple values for one header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .addHeader(headerName, headerValue1)
+            .addHeader(headerName, headerValue2)
+            .build()
+
+        // Expect
+        assertEquals(listOf(headerValue1, headerValue2), request.headers.values(headerName))
+    }
+
+    @Test
+    fun `should set one single header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue = "application/json"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .setHeader(headerName, headerValue)
+            .build()
+
+        // Expect
+        assertEquals(headerValue, request.headers.get(headerName))
+    }
+
+    @Test
+    fun `should set multiple values for one header`() {
+        // Given
+        val method = Method.GET
+        val url = "https://example.com"
+        val headerName = "content-type"
+        val headerValue1 = "application/json"
+        val headerValue2 = "text/plain"
+
+        // When
+        val request = Request
+            .builder()
+            .method(method)
+            .url(url)
+            .setHeader(headerName, headerValue1)
+            .setHeader(headerName, headerValue2)
+            .build()
+
+        // When
```

Comment:
Sorry for the nitpicking, you missed updating this.

Reply:
Fixed!